staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -267,6 +294,9 @@ func (a *Authenticator) getCreds() (*credentials, error) {
 	if err := a.refreshCredsLocked(nil); err != nil {
 		return nil, err
 	}
+
+	// Update the shortest client TTL from cache with the latest time.
+	metrics.ClientCertTTL.Set(globalCache.shortestTTL(time.Now))
```

Comment:
Since we changed it from _expiry_ to _TTL_, it needs to be frequently set or else the value will be super out of date.

If it is only set on refreshCredsLocked it will initially set to say _4 years_, and the TTL will always read as 4 years until it eventually rotated weeks-years later.

Reply:
moving it into refreshCredsLocked doesn't change how frequently it's called, right?

Reply:
I've moved it into a `defer` in `getCreds()` which catches all.